dotnet-logging/LogLevelsWithSerilog/Tests/HomeControllerTests.cs

```diff
@@ -0,0 +1,112 @@
+using AutoFixture;
+using LogLevelsWithSerilog.Controllers;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace LogLevelsWithSerilogTests;
+
+[TestClass]
+public class LogLevelsWithSerilogTests
+{
+    private HomeController? _homeController;
+    private Mock<ILogger<HomeController>>? _loggerMock;
+    private IFixture? _fixture;
+
+    [TestInitialize]
+    public void Setup()
+    {
+        _fixture = new Fixture();
+        _loggerMock = new Mock<ILogger<HomeController>>();
+        _homeController = new HomeController(_loggerMock.Object);
+    }
+
+    [TestMethod]
+    public void WhenLogRouteCalledWithDebug_ThenDebugLogMustBeCalled()
+    {
+        var message = _fixture.Create<string>();
+        var logType = (int)LogLevel.Debug;
+
+        _homeController?.GenerateLog(message, logType);
+
+        _loggerMock?.Verify(x => x.Log(LogLevel.Debug,
+            It.IsAny<EventId>(),
+            It.Is<It.IsAnyType>((v, t) => true),
+            It.IsAny<Exception>(),
+            It.Is<Func<It.IsAnyType, Exception?, string>>((v, t) => true)), Times.Once);
```

Comment:
I generated a private "TestGenerateLog()" method because testing scenario was similar for the test methods. Inside TestGenerateLog() method, I added exact match control. For each of my test methods, I called TestGenerateLog() with appropriate log level to run test.